lib/block_view_validator.go

```diff
@@ -1924,6 +1924,36 @@ func (bav *UtxoView) GetGlobalActiveStakeAmountNanos() (*uint256.Int, error) {
 	return globalActiveStakeAmountNanos, nil
 }
 
+func (bav *UtxoView) ShouldJailValidator(validatorEntry *ValidatorEntry) (bool, error) {
+	// Return false if the validator is already jailed. We do not want to jail
+	// them again as we want to retain their original JailedAtEpochNumber so
+	// that they can eventually unjail themselves.
+	if validatorEntry.Status() == ValidatorStatusJailed {
+		return false, nil
+	}
+
+	// Retrieve the CurrentEpochNumber.
+	currentEpochNumber, err := bav.GetCurrentEpochNumber()
```

Comment:
how do we know if they satisfied some criteria such that they shouldn't be jailed? Seems like we'd end up with every validator being jailed as it stands?

Reply:
That is correct as it stands now. I just updated the description above with this line:
> A validator stays active by either voting on blocks or proposing new blocks. When they participate in consensus by performing either of those actions, we update their LastActiveAtEpochNumber.
We haven't implemented the consensus logic to handle VOTES or new BLOCK_PROPOSALS. But in that logic, we will be updating the `ValidatorEntry.LastActiveAtEpochNumber = CurrentEpochNumber`.